src/python/pants/engine/goal.py

```diff
@@ -44,69 +30,63 @@ def deprecated_cache_setup_removal_version(cls):
     """
     return None
 
+  @classproperty
+  def options_scope(cls):
+    return cls.name
+
   @classmethod
-  def register_options(cls, register):
-    """Register options for the `Goal.Options` of this `Goal`.
+  def subsystem_dependencies(cls):
+    # NB: `Goal.Options` implements `SubsystemClientMixin` in order to allow v1 `Tasks` to
+    # depend on v2 Goals, and for `Goals` to declare a deprecated dependency on a `CacheSetup`
+    # instance for backwards compatibility purposes. But v2 Goals should _not_ have subsystem
+    # dependencies: instead, the @rules participating (transitively) in a Goal should directly
+    # declare their Subsystem deps.
+    if cls.deprecated_cache_setup_removal_version:
+      dep = CacheSetup.scoped(
+        cls,
+        removal_version=cls.deprecated_cache_setup_removal_version,
+        removal_hint='Goal `{}` uses an independent caching implementation, and ignores `{}`.'.format(
+          cls.options_scope,
+          CacheSetup.subscope(cls.options_scope),
+        )
+      )
+      return dep,
+    return tuple()
+
+  options_scope_category = ScopeInfo.GOAL
+
+  def __init__(self, scope, scoped_options):
+    # NB: This constructor is shaped to meet the contract of `Optionable(Factory).signature`.
+    super().__init__()
+    self._scope = scope
+    self._scoped_options = scoped_options
+
+  @property
+  def values(self):
+    """Returns the option values for these Goal.Options."""
+    return self._scoped_options
+
+
+@dataclass(frozen=True)
+class Goal:
+  """The named product of a `@console_rule`.
 
-    Subclasses may override and call register(*args, **kwargs). Callers can retrieve the resulting
-    options values by declaring a dependency on the `Goal.Options` class.
-    """
+  This abstract class should be subclassed and given a `Goal.name` that it will be referred to by
```

Comment:
These two lines are now more descriptive of the `GoalSubsystem` than of the `Goal`